crypto/src/main/java/org/hyperledger/besu/crypto/EllipticCurveSignature.java

```diff
@@ -0,0 +1,43 @@
+package org.hyperledger.besu.crypto;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.bouncycastle.math.ec.ECPoint;
+
+import java.math.BigInteger;
+import java.util.Optional;
+import java.util.function.UnaryOperator;
+
+public interface EllipticCurveSignature {
```

Comment:
I feel we should go a bit further than simply pulling out the interface methods.  I think we should reconsider all the APIs being pulled through and look to see if better homes exist (calculateECDHKeyAgreement, getHalfCureOrder, enableNarive, etc).

A higher level goal may be to provide SPI classes to make it work with javax.crypto and friends, rather than a single use interface.

I also think we can do better than the `EllipticCurveSignature` name.  Account sigs may not always be ECDSA (DSA and RSA are also listed in FIPS). So perhaps `SignatureAlgorithm`?  Or make it role based `AccountSignatures`.  Also the noun implies it is the signature itself, not the class that does the signing, verification, and key management.

Reply:
> I feel we should go a bit further than simply pulling out the interface methods. I think we should reconsider all the APIs being pulled through and look to see if better homes exist (calculateECDHKeyAgreement, getHalfCureOrder, enableNarive, etc).

I believe the idea is to make incremental changes. And because this class is used pretty much everywhere, pulling out the interface already introduces a bunch of changes to review. Do you feel like it is better to have all changes done at once?

> I also think we can do better than the EllipticCurveSignature name. Account sigs may not always be ECDSA (DSA and RSA are also listed in FIPS)

I think supporting "non-EC" have bigger implications that are out of the scope of this PR. I guess this is a small step towards something like https://github.com/EntEthAlliance/client-spec/issues/110. (a really small step)